drivers/ethernet/Kconfig.nxp_s32_netc

```diff
@@ -1,11 +1,10 @@
-# Copyright 2022 NXP
+# Copyright 2022-2023 NXP
 # SPDX-License-Identifier: Apache-2.0
 
 menuconfig ETH_NXP_S32_NETC
 	bool "NXP S32 Ethernet Switch and Controller (NETC) driver"
 	default y
 	depends on (DT_HAS_NXP_S32_NETC_PSI_ENABLED || DT_HAS_NXP_S32_NETC_VSI_ENABLED)
-	depends on !NET_TEST
```

Comment:
This should work now after #64180 is merged.

Reply:
Nope. The issue that PR address is when run tests in tests\net, the default network interface is used is the dummy one.
But the `tests/drivers/build_all/ethernet` is enabling `CONFIG_NET_L2_ETHERNET`. Not sure i should touch on the test to set `CONFIG_NET_TEST=n` or not since it has been there for a few year. Also, there is no other ethernet shim driver is adding `depends on !NET_TEST` in Kconfig, so i decided to remove this one

Reply:
Good if the dummy was selected as a default before the PR I mentioned. Please note that `CONFIG_NET_TEST` must be set for all the tests under `tests/net/` directory proper things are done in other parts of the system.

I am not saying you should keep the "depends on", just noting that there was an issue for boards that have built-in Ethernet and when tests were run in that board (and that issue should be fixed by the PR I mentioned).

Reply:
Right, i agree. The PR you mentioned should fixed the issue about built-in Ethernet and `tests/net`, `tests/net` is working fine on our platform